Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugMultiplayerMetrics.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Debug/MultiplayerDebugMultiplayerMetrics.h>
+#include <Multiplayer/IMultiplayer.h>
+
+namespace Multiplayer
+{
+#ifdef IMGUI_ENABLED
+    void AccumulatePerSecondValues(const MultiplayerStats& stats, const MultiplayerStats::Metric& metric, float& outCallsPerSecond, float& outBytesPerSecond)
+    {
+        uint64_t summedCalls = 0;
+        uint64_t summedBytes = 0;
+        for (uint32_t index = 0; index < MultiplayerStats::RingbufferSamples; ++index)
+        {
+            summedCalls += metric.m_callHistory[index];
+            summedBytes += metric.m_byteHistory[index];
+        }
+        const float totalTimeSeconds = static_cast<float>(stats.m_totalHistoryTimeMs) / 1000.0f;
+        outCallsPerSecond += (summedCalls > 0 && totalTimeSeconds > 0.0f) ? aznumeric_cast<float>(summedCalls) / totalTimeSeconds : 0.0f;
+        outBytesPerSecond += (summedBytes > 0 && totalTimeSeconds > 0.0f) ? aznumeric_cast<float>(summedBytes) / totalTimeSeconds : 0.0f;
```

Comment:
Can totalTimeSeconds be zero?

Reply:
Yup, it's possible on the very first frame, hence the ternary to avoid a float divide by zero crash.